### ISSUE TITLE
remove usages of `Task`, `IO`, `RIO`, `UIO`, `URIO` from ZIO codebase and documentation

### ISSUE THREAD

**Issue by domdorn** [zio/zio#6931]
![grafik](https://user-images.githubusercontent.com/100349/175048243-fd7f8a3d-13c6-47e9-aa60-ee78a5c3fb5f.png)

The type aliases will stay in the codebase (for legacy reasons) but should not be used in other parts of the codebase or in official zio documentation.

**Comment by adamgfraser** [comment]
I tend to agree. I have moved away from using the type aliases myself but there is a risk of being "preachy" here.

I also worry that we have people who like them, people who don't, and we're kind of using the codebase and documentation as a battleground between that resulting in compromises such as "we should have this feature but it shouldn't be documented" or "we should have this coding guideline that you shouldn't use this feature even though it is not deprecated and precisely fits your use case" which is going to be hard to maintain.

**Comment by jdegoes** [comment]
I don't dislike them so much as to feel strongly about this issue.

But I have observed that some users new to ZIO struggle with memorizing the multitude of (inconsistent) type aliases (or with reading ZIO documentation and encountering type aliases they haven't memorized); I have also observed that, in general (not specific to ZIO), type aliases (rather than newtypes) add one layer of indirection and make it more difficult to understand what code is doing. 

But separately from all of this, I think a small and stable set of type aliases can accelerate productivity for experienced users (bottom of the funnel). Ultimately I view it as top-of-funnel versus bottom-of-funnel optimization problem.

**Comment by guizmaii** [comment]
Agreed with @ghostdogpr.

On another note, sorry if it's a bit out of the scope of this issue but, IMO, there are more important things to remove from ZIO documentation to help beginners.
IMO, the first thing to remove from it are the references and examples of the "ZIO service pattern v1" and to replace these examples with the "ZIO service pattern v2".

I'm recruiting a lot these days. During the recruitment process, I ask the candidates to write a small app with ZIO. 
Most of the candidates are new to ZIO, even if not all of them are new to FP.
Most of these candidates, even experienced engineers, are using the cumbersome and weird "ZIO service pattern v1" just because it's what they've seen in the documentation.

They've lost time to understand and use something that we all, I believe, consider as deprecated.

I haven't traversed all the doc, but was quickly able to find some references to this pattern here for example: https://zio.dev/version-1.x/howto/mock-services

**Comment by domdorn** [comment]
the big question (imho) is: 
**"What is the long-term goal of ZIO?"** 

I thought from reading between the lines in Malaga and elsewhere that we want to **be able to put a java dev / scala newbie in front of a project and they'd be able to figure out how this "zio thing" works** from having a basic scala knowledge, reading a little documentation and code... not using the aliases in zio-core code and the accompaning scaladoc would IMHO help in this case.. why?

e.g. see this screenshot (still zio 1) from a hover of `ZIO.effect`:
![grafik](https://user-images.githubusercontent.com/100349/175106198-14e14ac9-bfd3-4037-870d-b4f6fecdf6a7.png)
it returns a `Task[A]`.. someone new to ZIO now has to _learn_/memorize the concept of `Task` (and the other aliases) .. a CMD+click (on osx) will bring them to 
![grafik](https://user-images.githubusercontent.com/100349/175109725-9e1069e0-308c-4e02-907d-688ce131631d.png)
.. still no clue what `Task` is.. another CMD + click and they're finally at 
![grafik](https://user-images.githubusercontent.com/100349/175109835-046cd7b2-661c-4aa7-9d26-4670e0d195f2.png)
explaining that `Task` is a effect that might fail with a `Throwable` or succeed with a `A` ..  

Why they have to learn this? Cause ZIO has some history with ScalaZ and we're too afraid to break ties with ScalaZ and Haskell cause some (advanced/senior) users are accustomed to those aliases.

The thing is, **someone advanced/senior with ZIO does not have a problem with `Task`/`UIO`/`RIO` etc.** They also don't have a problem with a `ZIO[Any, Throwable, A]` for that matter. But **someone already struggling to understand everything of "this ZIO thing" might appreciate a little less complexity**.

**Comment by luis3m** [comment]
I agree with @ghostdogpr as well.

Personally, I find the implied cost of learning / memorising these type aliases inflated.

**Comment by adamgfraser** [comment]
@domdorn I think the thing is you can take this in two different directions.

The first is you can not use `Task`, in which case you're right you don't need to learn the name of a type alias.

But the other is `Task` should be the first thing you use. You're learning a lot of other concepts like the idea that all of these workflows are just descriptions. Three type parameters is a lot! What are all of them? In that case `Task` is potentially the easiest "first step" for the Java developer, allowing them to introduce the concepts of typed errors and environmental dependencies as they learn them.

**Comment by domdorn** [comment]
hmm... 
- I had two coworkers new to fp/zio.. both of them sticked a cheat-sheet/post-it with the mapping of these types to the frame of their screen... for me this is/was enough to see that these types are not "beginner friendly" .. 
- in talks/presentations about zio for beginners we have a slide about these types that has to be shown & explained _in every presentation_.. we show them for ~45-60 seconds, then we hope/asume people have memorized them for the next 45-60 minutes.. if they haven't or forgotten them after a few minutes or mixed them up, we risk losing them during the talks.. 
- in a in-person talk or workshop I'd give out handouts with this cheatsheet (https://github.com/ghostdogpr/zio-cheatsheet/blob/master/README.md)  so people can look them up.. 
- when I started with zio, I had to buy zionomicon to be able to understand what was going on (coming from a cats `EitherT` world) and in later chapters had to jump back to the chapter with these type mappings till I realized "maybe its a good idea to write these down on paper to have available when reading the rest of the book".. 


We put a lot of effort into making our APIs discoverable by putting operations in companion objects, having naming conventions, service patterns, not having `Has` anymore, only one method to do a certain thing instead of ten, no implicits etc. .. yet we force people to memorize some aliases/types (again 5 instead of 1) that are only here cause they where used by libraries our users hopefully have never to use (scalaz, cats-effect) or languages that where once inspirational to scala but their ecosystem is years behind what we accomplish with ZIO on the JVM.... 

I don't see how this aligns with commercial applicability of FP and bringing this to a broad group of users. 
I obviously have a strong opinion about this. In my opinion these types are a relict of the past and make it unnecessarily hard for new users to adopt ZIO and Scala. 

> But the other is Task should be the first thing you use. You're learning a lot of other concepts like the idea that all of these workflows are just descriptions. Three type parameters is a lot! What are all of them? In that case Task is potentially the easiest "first step" for the Java developer, allowing them to introduce the concepts of typed errors and environmental dependencies as they learn them.

Then we maybe should not have removed the companion object of `Task`. Why does. a `ZIO.attempt` returns anything else than a `ZIO`?

As for teaching: this is enough for users to grasp the `E` and `A` of ZIO.. 
```scala
import zio._
import java.io.IOException

object Sample {


  val effectThatReturnsUnit: ZIO[Any, Nothing, Unit] = ZIO.debug("Hello World")

  val effectThatReturnsAString: ZIO[Any, Nothing, String] = ZIO.succeed("Hello World")

  val effectThatCombines: ZIO[Any, Nothing, Unit] = effectThatReturnsAString.flatMap(s => ZIO.debug(s))

  val effectThatMayFail: ZIO[Any, Throwable, String] = zio.Console.readLine


  val effectReadingAndPrinting: ZIO[Any, Throwable, Unit] = effectThatMayFail.flatMap(input => zio.Console.printLine(input))

  val effectWithMoreConcreteException: ZIO[Any, IOException, String] = effectThatMayFail.refineToOrDie[IOException]

  def main(args: Array[String]) = {
   // run the whole thing
    zio.Runtime.default.unsafeRun(effectReadingAndPrinting)
  }

}
```

In the end - while having a strong opinion - its not up to me to decide this. I'm just saying that these type aliases make it unnecessarily complicated for new users, especially in the first few hours and days when they are already overwhelmed with other concepts they have to grasp. 
If we are fine with having complicated stuff, we should think about dramatically improving the documentation and making zionomicon freely available (like the cats book). (IMHO) Our documentation is years behind the levels that are normal in other comercial oss-projects, e.g. spring-framework and weeks behind "non-commercial" ones like cats. I thought we'd compensate this with an API that is just simple and straightforward to use, so documentation is almost never needed. I know there are already serious efforts going into improving the documentation and I highly appreciate that! 
Imho we don't have to decide on this before the 2.0 release. Refactoring the code base (scalafix or a regexp search+replace) can also happen later. What we should decide on is if we want to keep teaching these concepts. If yes, we have to drastically improve on the documentation and make sure that its easy for users to understand the code that is displayed (e.g. by providing functionality that provides a tooltip when the user hovers over a `Task[String]` that then shows the mapping table or have it visible on every page (we can't assume they read the documentation from beginning to end but will jump in randomly from a google search) until the user decides they don't need it anymore by closing it explicitely.

**Comment by khajavi** [comment]
> IMO, the first thing to remove from it are the references and examples of the "ZIO service pattern v1" and to replace these examples with the "ZIO service pattern v2".

Currently, in ZIO 2.x documentation we only teach one pattern to write services. If you found any problems according to this, let me know to fix that.
